alerting/alerting-jar/src/main/java/org/hawkular/metrics/alerting/InsertedDataSubscriber.java

```diff
@@ -86,100 +91,85 @@
     @Configurable
     @ConfigurationProperty(DISABLE_PUBLISH_FILTERING)
-    private String disablePublishFiltering;
+    private String disablePublishFilteringProperty;
+    private boolean disablePublishFiltering;
 
     public void onMetricsServiceReady(@Observes @ServiceReady ServiceReadyEvent event) {
-        if (!Boolean.parseBoolean(disableMetricsForwarding)) {
+        if (!Boolean.parseBoolean(disableMetricsForwardingProperty)) {
             publishPeriod = getPublishPeriod();
+            disablePublishFiltering = Boolean.parseBoolean(disablePublishFilteringProperty);
             Observable<List<Metric<?>>> events;
-            if (Boolean.parseBoolean(disablePublishFiltering)) {
-                events = event.getInsertedData()
-                        .buffer(publishPeriod, TimeUnit.MILLISECONDS, 100)
-                        .filter(list -> !list.isEmpty());
-            } else {
-                events = event.getInsertedData()
-                        .filter(m -> m.getType() != MetricType.STRING)
-                        .buffer(publishPeriod, TimeUnit.MILLISECONDS, 100)
-                        .filter(list -> !list.isEmpty());
-            }
+            events = event.getInsertedData()
+                    .buffer(publishPeriod, TimeUnit.MILLISECONDS, 100)
+                    .filter(list -> !list.isEmpty());
             events = events.onBackpressureBuffer()
                     .observeOn(Schedulers.io());
             subscription = events.subscribe(list -> onInsertedData(list));
         }
     }
 
     private void onInsertedData(List<Metric<?>> metrics) {
-        List<Data> data = new ArrayList<>(metrics.size());
-        metrics.stream().forEach(m -> addToData(m, data));
-        if (!data.isEmpty()) {
+        List<Data> dataToSend = new ArrayList<>(metrics.size());
+        metrics.stream().forEach(m -> addToData(m, dataToSend));
+        if (!dataToSend.isEmpty()) {
             try {
-                log.tracef("Sending: %s", data);
-                alertsService.sendData(data);
+                alertsService.sendData(dataToSend);
 
             } catch (Exception e) {
                 log.warnf("Failed to send alerting data.", e);
             }
         }
     }
 
+    private boolean containsKey(String tenantId, String dataId) {
+        CacheKey cacheKey = new CacheKey(tenantId, dataId);
+        return dataIdCache.containsKey(cacheKey);
+    }
+
     @SuppressWarnings("unchecked")
     private void addToData(Metric<?> metric, List<Data> data) {
-        MetricType<?> metricType = metric.getMetricId().getType();
+        MetricId<?> metricId = metric.getMetricId();
+        MetricType<?> metricType = metricId.getType();
         if (metricType == MetricType.UNDEFINED) {
             return;
         }
 
+        String tenantId = metricId.getTenantId();
+        String dataId = prefixMap.get(metricType) + metricId.getName();
+        if (!disablePublishFiltering && !containsKey(tenantId, dataId)) {
```

Comment:
Glad to see this change. Not creating objects for all incoming data removes the memory pressure we've seen recently.